src/asteroids/AsteroidController.js

```diff
@@ -0,0 +1,117 @@
+import LargeAsteroid from "./LargeAsteroid.js";
+import MediumAsteroid from "./MediumAsteroid.js";
+import SmallAsteroid from "./SmallAsteroid.js";
+
+//Asteroid Controller is responsible for generating and clearing waves of asteroids.
+class AsteroidController {
+  //In order to prevent any null errors we need to init this.asteroids with values.
+  genAsteroids(scene, num) {
+    for (let i = 0; i < num; i++) {
+      this.spawnAsteroid(scene);
+    }
+  }
+
+  //Deletes all asteroids from the field
+  clearAsteroids() {
+    for (let i = 0; i < this.asteroids.length; i++) {
+      this.asteroids[i].destroy();
+    }
+  }
+
+  spawnAsteroid(scene) {
+    //Determine what type of asteroid to spawn
+    let asteroidType = Math.floor(Math.random() * 3);
+    //Generate what side to spawn the asteroid from (left,right,top,bottom)
+    let side = Math.floor(Math.random() * 4);
+    //Using the side, generate the x,y values where the asteroid is supposed to spawn
+    let spawnLocation = this.genSpawnLocation(
+      side,
+      scene.worldWidth,
+      scene.worldHeight
+    );
+    //Using the side, generate the direction the asteroid should be moving
+    let direction = this.genDirection(side);
+    if (asteroidType === 0) {
+      //If 0 spawn Large Asteroid
+      return new LargeAsteroid(
+        scene,
+        spawnLocation.x,
+        spawnLocation.y,
+        direction,
+        Math.floor(Math.random() * 100) + 50
+      );
+    }
+    //If 1 spawn Medium Asteroid
+    else if (asteroidType === 1) {
+      return new MediumAsteroid(
+        scene,
+        spawnLocation.x,
+        spawnLocation.y,
+        direction,
+        Math.floor(Math.random() * 200) + 50
+      );
+    }
+    //If 2 spawn small Asteroid
+    else if (asteroidType === 2) {
+      return new SmallAsteroid(
+        scene,
+        spawnLocation.x,
+        spawnLocation.y,
+        direction,
+        Math.floor(Math.random() * 300) + 50
+      );
+    }
+  }
+
+  //Determines the x,y values for where the asteroid is supposed to spawn.
+  genSpawnLocation(side, gameWidth, gameHeight) {
+    //left
+    if (side === 0) {
+      return { x: 0, y: this.randomIntFromInterval(0, gameHeight) };
+    }
+    //right
+    else if (side === 1) {
```

Comment:
For the 0,1,2,3, it'd be nice if these were defined above the class as an enum. Then you can have side === LEFT, where LEFT = 0.